### ISSUE TITLE
App folder: styles applied after delay on prefetched pages

### ISSUE THREAD

**Issue by Arctomachine** [vercel/next.js#46754]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.13.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.2.4-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://github.com/Arctomachine/next-prefetch-css-modules-bug

### To Reproduce

1. `next build` && `next start`
2. Navigate to /posts page

### Describe the Bug

Prefetched pages are not applying styles instantly after transition. At first only raw html is loaded without any styles. Then styles are loaded and applied. It makes visible flickering on such pages.
It happens with css modules, as style files for modules are only loaded when their respective components are accessed. But since prefetched page already has required data, it can show page contents faster than styles are being loaded.
The further server is from user, the longer it takes for styles to be applied.

Hosted reproduction: https://next-prefetch-css-modules-bug.vercel.app/

### Expected Behavior

Transition to prefetched page should apply styles before showing page contents.

### Which browser are you using? (if relevant)

any

### How are you deploying your application? (if relevant)

next start

**Comment by Arctomachine** [comment]
Actually it works like that even on not prefetched pages. If the page is static, when it gets accessed for the first time from another page, it behaves like that. It loads text, and only then it loads and applies styles, which is clearly visible.

**Comment by timfee** [comment]
See https://github.com/vercel/next.js/issues/46947

**Comment by smitbarmase** [comment]
Yes. Happening with few pages. It's strange that this is only happening with few pages and not with others.

**Comment by marcoshernanz** [comment]
Thanks for the report. This was fixed by the Suspensey CSS work covered by #47675 and #47708, and the same reproduction was tracked in #46868. I also re-tested it on current stable and canary releases with delayed route CSS, with and without prefetching, and the page now waits for its styles before committing. Closing as fixed; if this still reproduces on a current release, please open a new issue with an updated repro.
